sysdig/getmetrics_sysdig.py

```diff
@@ -0,0 +1,348 @@
+    parser.add_option("-w", "--serverUrl",
+                      action="store", dest="serverUrl", help="Server Url")
+    parser.add_option("-c", "--chunkLines",
+                      action="store", dest="chunkLines", help="Timestamps per chunk for historical data.")
+    parser.add_option("-l", "--logLevel",
+                      action="store", dest="logLevel", help="Change log verbosity(WARNING: 0, INFO: 1, DEBUG: 2)")
+    (options, args) = parser.parse_args()
+
+    params = dict()
+    if options.serverUrl is None:
+        params['serverUrl'] = 'http://stg.insightfinder.com'
+    else:
+        params['serverUrl'] = options.serverUrl
+    if options.chunkLines is None:
+        params['chunkLines'] = 50
+    else:
+        params['chunkLines'] = int(options.chunkLines)
+    params['logLevel'] = logging.INFO
+    if options.logLevel == '0':
+        params['logLevel'] = logging.WARNING
+    elif options.logLevel == '1':
+        params['logLevel'] = logging.INFO
+    elif options.logLevel >= '2':
+        params['logLevel'] = logging.DEBUG
+
+    return params
+
+#
+# Read and parse InsightFinder config from config.ini
+#
+
+
+def get_agent_config_vars():
+    if os.path.exists(os.path.abspath(os.path.join(__file__, os.pardir, "config.ini"))):
+        config_parser = ConfigParser.SafeConfigParser()
+        config_parser.read(os.path.abspath(os.path.join(__file__, os.pardir, "config.ini")))
+        try:
+            user_name = config_parser.get('insightfinder', 'user_name')
+            license_key = config_parser.get('insightfinder', 'license_key')
+            project_name = config_parser.get('insightfinder', 'project_name')
+            sampling_interval = config_parser.get('insightfinder', 'sampling_interval')
+            if_http_proxy = config_parser.get('insightfinder', 'if_http_proxy')
+            if_https_proxy = config_parser.get('insightfinder', 'if_https_proxy')
+
+            # host_chunk_size = int(config_parser.get('insightfinder', 'host_chunk_size'))
+            # metric_chunk_size = int(config_parser.get('insightfinder', 'metric_chunk_size'))
+        except ConfigParser.NoOptionError:
+            logger.error(
+                "Agent not correctly configured. Check config file.")
+            sys.exit(1)
+
+        if len(user_name) == 0:
+            logger.warning(
+                "Agent not correctly configured(user_name). Check config file.")
+            sys.exit(1)
+        if len(license_key) == 0:
+            logger.warning(
+                "Agent not correctly configured(license_key). Check config file.")
+            sys.exit(1)
+        if len(project_name) == 0:
+            logger.warning(
+                "Agent not correctly configured(project_name). Check config file.")
+            sys.exit(1)
+
+        config_vars = {
+            "userName": user_name,
+            "licenseKey": license_key,
+            "projectName": project_name,
+            "samplingInterval": sampling_interval,
```

Comment:
should be required or have a default

Reply:
It should be required. Made changes to check it its empty.